docs/examples/Naive_Classification.py

```diff
@@ -0,0 +1,188 @@
+# which just says whether the target is small, medium or large. Similar to the state space
+# categories, this is represented by a detection state vector with 3 components, either 0 or 1.
+# To determine which classification is observed, we randomly sample from the multinomial
+# distribution defined by the row of the emission matrix corresponding to the most likely state of
+# the target.
+#
+# .. math::
+#       E = \begin{bmatrix}
+# 		P(Small | Bicycle) & P(Medium | Bicycle) & P(Large | Bicycle)\\
+# 		P(Small | Car) & P(Medium | Car) & P(Large | Car)\\
+# 		P(Small | Bus) & P(Medium | Bus) & P(Large | Bus)
+# 	    \end{bmatrix}
+#
+
+import scipy
+
+E = np.array([[0.89, 0.1, 0.01],
+              [0.3, 0.3, 0.4],
+              [0.1, 0.3, 0.6]])
+
+# ie. 89% of bicycles are small, 10% are medium, 1% are large.
+
+
+def observe(state):
+    x = state.state_vector
+
+    row_num = np.argmax(x)
+
+    row = E[row_num]
+
+    sample = _sample(row)
+
+    return StateVector(sample)
+
+
+def _sample(row):
+    rv = scipy.stats.multinomial(n=1, p=row)
+    return rv.rvs(size=1, random_state=None)
+
+
+# %%
+
+from stonesoup.types.detection import Detection
+
+observations = []
+for i in range(0, nsteps):
+    observations.append(Detection(observe(gt[i])))
+
+
+# %%
+# Tracking classification
+# -----------------------
+# Posterior state is given by :math:`X_{t} = EZ_{t} * T^T X_{t-1}`, where :math:`X_{t}` is the
+# state estimate at time :math:`t`, :math:`E` the emission matrix, :math:`Z_{t}` an observation at
+# time :math:`t` and :math:`T` the transition model matrix, where :math:`*` notates piecewise
+# vector multiplication.
+# We begin with a prior state guess, whereby we have no knowledge of the target's classifcaiton,
+# and therefore appoint equal probability to each category.
+
+prior = State(StateVector([1/3, 1/3, 1/3]))
+
+# %%
+# Next, we carry-out the tracking loop, making sure to normalise the posterior estimate at each
+# iteration, as track state vector components represent a categorical distribution (they must
+# therefore sum to 1).
+
+from stonesoup.types.track import Track
+
+track = Track()
+for observation in observations:
+    TX = transit(prior)
+    EY = E @ observation.state_vector
+
+    prenormalise = np.multiply(TX, EY)
+
+    normalise = prenormalise / np.sum(prenormalise)
+
+    track.append(State(normalise))
+    prior = track[-1]
+
+
+# %%
+# Plotting
+# --------
+# We plot the track classification as a stacked bar graph. Green indicates bicycle, white car, and
+# red bus. The larger a particular bar, the greater the probability that the track at that time
+# is of the corresponding classification.
+# Observations are plotted underneath this graph. Light blue indicates an observation that the
+# target is small, blue that it is medium, and dark blue that it is large.
+
+import matplotlib.pyplot as plt
+
+t_a, t_n, t_e = np.array([list(state.state_vector) for state in track]).T
```

Comment:
Just wondering - what do the a, n and e in  `t_a, t_n, t_e` stand for?

Reply:
Just some very bad variable naming from me. Since Id changed the original class names, they don't mean anything now